eZ/Bundle/EzPublishCoreBundle/Command/RegenerateUrlAliasesCommand.php

```diff
@@ -0,0 +1,216 @@
+                'bulk-count',
+                'c',
+                InputOption::VALUE_OPTIONAL,
+                'Number of Locations fetched into memory and processed at once',
+                50
+            )
+            ->addOption(
+                'user',
+                'u',
+                InputOption::VALUE_OPTIONAL,
+                'eZ Platform user used when running this command',
+                'admin'
+            )
+            ->setHelp(
+                <<<EOT
+The command <info>%command.name%</info> regenerates URL aliases for Locations and cleanups
+corrupted URL aliases (pointing to non-existent Locations).
+Existing aliases are archived (will redirect to the new ones).
+
+<error>During the script execution the database should not be modified.</error>
+
+Since this script can potentially run for a very long time, to avoid memory exhaustion run it in
+production environment using <info>--env=prod</info> switch.
+
+EOT
+            );
+    }
+
+    /**
+     * @param \Symfony\Component\Console\Input\InputInterface $input
+     * @param \Symfony\Component\Console\Output\OutputInterface $output
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException
+     */
+    protected function initialize(InputInterface $input, OutputInterface $output)
+    {
+        $this->permissionResolver->setCurrentUserReference(
+            $this->userService->loadUserByLogin($input->getOption('user'))
+        );
+    }
+
+    /**
+     * Regenerate URL aliases.
+     *
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $bulkCount = (int)$input->getOption('bulk-count');
+
+        $query = new LocationQuery(
+            [
+                'filter' => new Subtree('/1/'),
+            ]
+        );
+        $pager = new Pagerfanta(
+            new LocationSearchAdapter($query, $this->searchService)
+        );
+        $pager->setMaxPerPage($bulkCount);
+
+        $output->writeln(
+            sprintf(
+                '<info>Found %d Locations. Regenerating System URL aliases...</info>',
+                $pager->getNbResults()
+            )
+        );
+
+        $progressBar = $this->getProgressBar($pager->getNbResults(), $output);
+        $progressBar->start();
+
+        $pagesCount = $pager->getNbPages();
+        for ($currentPage = 1; $currentPage <= $pagesCount; ++$currentPage) {
+            $pager->setCurrentPage($currentPage);
+            $this->processLocations($pager, $progressBar, $output);
+        }
+        $progressBar->finish();
+        $output->writeln('');
+
+        $output->writeln('<info>Cleaning up corrupted URL aliases...</info>');
+        $corruptedAliasesCount = $this->urlAliasService->deleteCorruptedUrlAliases();
+        $output->writeln("<info>Done. Deleted {$corruptedAliasesCount} entries.</info>");
+    }
+
+    /**
+     * Return configured progress bar helper.
+     *
+     * @param int $maxSteps
+     *
```

Comment:
not needed empty line

Reply:
Fixed via f96dd03 along with some other fixes.